apistar/components/dependency.py

```diff
@@ -450,3 +467,26 @@ def container_type(self,
             detail = str(exc)
 
         raise exceptions.ValidationError(detail=detail)
+
+
+InjectorState = typing.NewType('InjectorState', dict)
+
+
+class InjectionCaller:
+
+    def __init__(self, injector: Injector, state: InjectorState) -> None:
+        self.injector = injector
+        self.state = state
+        self.is_async = isinstance(injector, AsyncDependencyInjector)
+
+    def __call__(self, func):
+        if asyncio.iscoroutinefunction(func):
+            return self.run_async(func)
```

Comment:
This doesn't quite work out, since we actually want to use `await` on this, and call it from `async` functions only.